### DISCUSSION TITLE
Сохранение паролей из формы в таблицу Пользователи

### DISCUSSION THREAD

**Discussion by gusin-itas** [totumonline/totum-ru-issues-and-discussions#426] [Как это сделать?] [answered]
Требуется занести логин и пароль нового пользователя из формы регистрации в таблицу Пользователи. 

Выполняю для строки пароля из формы код действия по переносу пароля в таблицу Пользователей:
"= : insert(table: 'users'; field: 'login' = #name ; field: 'pass' = $passenc; field: 'email' = #email; log: true)
passenc: strEncrypt(str: #password)"

Для использования функции strEncrypt создал в каталоге /home/totum/totum-mit файл Crypto.key. В качестве содержимого файла сделал строку "totum". Есть ли вообще разница, что внутри этого файла находится или туда нужно что-то особым способом загонять? Нигде не нашёл про это информации...

Новая запись в таблице Пользователи появляется, но новый пользователь не может войти в Тотум с этим паролем. При ручном внесении новой записи в таблицу Пользователи всё работает правильно.

Что я делаю не так?

**Comment by poluninatm** [comment]
Пароли пользователей в таблице Пользователи хранятся в виде хешей md5.
Вам нужна функция strMd5

Вс, 6 авг. 2023 г. в 11:16, gusin-itas ***@***.***>:

**Reply by ttm-alexeypolunin** [reply, marked as answer]
Я передаю через set нешифрованную предварительно строку, мне кажется в этом случае она переводится в md5 автоматически. Если записывать напрямую в бд, тогда нужно md5.
